### PR TITLE
Adding ability to run adb reverse in mobly, fixes #900

### LINKED ISSUE #900: Add abilty to run 'adb reverse' via Mobly
We have the ability to run `adb forward` (for TCP port forwarding) but not `adb reverse` via mobly for TCP port reversing. This feature request is to add this ability in Mobly framework.

### PR DESCRIPTION
**Pull request by ssachinsunder**
Adding `AdbProxy.reverse()` method to run `adb reverse` via Mobly framework.

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/mobly/901)

### CODE REVIEW

**Review by mhaoli — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @ssachinsunder and @xpconanfan)

___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 634 at r1](https://reviewable.io/reviews/google/mobly/901#-NhV8HwYB8x2x1hGJxdg:-NhV8HwYB8x2x1hGJxdh:bj1ml5q) ([raw file](https://github.com/google/mobly/blob/c261e76ef1e98102fc00d1de1c6e5177b14992f2/tests/mobly/controllers/android_device_lib/adb_test.py#L634)):*
> ```Python
>   def test_reverse(self):
>     with mock.patch.object(adb.AdbProxy, '_exec_cmd') as mock_exec_cmd:
>       adb.AdbProxy().reverse(MOCK_SHELL_COMMAND)
> ```

Add an assertion to check that correct arguments are passed to _exec_cmd?



_Code quote:_
```Python
adb.AdbProxy().reverse(MOCK_SHELL_COMMAND)
```

**Comment by ssachinsunder** [comment]
Updated the `test_reverse()` and`test_forward()` with assert statements.

File: tests/mobly/controllers/android_device_lib/adb_test.py [hunk 1]
```diff
@@ -628,6 +628,22 @@ def test_getprops_when_empty_string_always_returned(self, mock_sleep):
...
+
+  def test_reverse(self):
+    with mock.patch.object(adb.AdbProxy, '_exec_cmd') as mock_exec_cmd:
+      adb.AdbProxy().reverse(MOCK_SHELL_COMMAND)
+      mock_exec_cmd.assert_called_with(['adb', 'reverse', 'ls'],
+                                       shell=False,
+                                       timeout=None,
+                                       stderr=None)
+
+  def test_reverse(self):
```
Comment (xpconanfan) [inline tests/mobly/controllers/android_device_lib/adb_test.py:644]: why are there two `test_reverse`?

**Review by xpconanfan — CHANGES_REQUESTED** [review]

File: tests/mobly/controllers/android_device_lib/adb_test.py (hunk 1 above)
Comment (ssachinsunder) [reply tests/mobly/controllers/android_device_lib/adb_test.py:644]: Oh. I think i made some mistake while committing the code. A little rusty on git. I'll fix it.

Comment (ssachinsunder) [reply tests/mobly/controllers/android_device_lib/adb_test.py]: Fixed it.

**Review by ssachinsunder — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mhaoli, @ssachinsunder, and @xpconanfan)

___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 634 at r1](https://reviewable.io/reviews/google/mobly/901#-NhV8HwYB8x2x1hGJxdg:-NhYlhbIB_uEpaDTBlgH:b-896fix) ([raw file](https://github.com/google/mobly/blob/c261e76ef1e98102fc00d1de1c6e5177b14992f2/tests/mobly/controllers/android_device_lib/adb_test.py#L634)):*
<details><summary><i>Previously, mhaoli (Minghao Li) wrote…</i></summary><blockquote>

Add an assertion to check that correct arguments are passed to _exec_cmd?
</blockquote></details>

Done.

**Review by mhaoli — APPROVED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: 1 of 2 files reviewed, 4 unresolved discussions (waiting on @ssachinsunder and @xpconanfan)

___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 630 at r3](https://reviewable.io/reviews/google/mobly/901#-Nh_j-BL2nfGW7wUARYr:-Nh_j-BL2nfGW7wUARYs:b-isjfcn) ([raw file](https://github.com/google/mobly/blob/61dac80e3b9b83d81b3fb783e6fdc1a865a81d06/tests/mobly/controllers/android_device_lib/adb_test.py#L630)):*
> ```Python
>   def test_forward(self):
>     with mock.patch.object(adb.AdbProxy, '_exec_cmd') as mock_exec_cmd:
>       adb.AdbProxy().forward(MOCK_SHELL_COMMAND)
> ```

Thanks for updating the test case test_forward. 

How about using args that are similiar to real forward command and reverse command in the test cases?

e.g.
adb.AdbProxy().forward(['tcp:12345', 'tcp:23456'])
adb.AdbProxy().reverse(['tcp:12345', 'tcp:23456'])



_Code quote:_
```Python
MOCK_SHELL_COMMAND
```
___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 638 at r3](https://reviewable.io/reviews/google/mobly/901#-Nh_jfiE0p6V_oO03lBJ:-Nh_jfiE0p6V_oO03lBK:b-lr0xth) ([raw file](https://github.com/google/mobly/blob/61dac80e3b9b83d81b3fb783e6fdc1a865a81d06/tests/mobly/controllers/android_device_lib/adb_test.py#L638)):*
> ```Python
>   def test_reverse(self):
>     with mock.patch.object(adb.AdbProxy, '_exec_cmd') as mock_exec_cmd:
>       adb.AdbProxy().reverse(MOCK_SHELL_COMMAND)
> ```

ditto here

**Review by ssachinsunder — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: 1 of 2 files reviewed, 4 unresolved discussions (waiting on @mhaoli and @xpconanfan)

___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 630 at r3](https://reviewable.io/reviews/google/mobly/901#-Nh_j-BL2nfGW7wUARYr:-NhbO97LB0Z_H3l4tgG9:b-896fix) ([raw file](https://github.com/google/mobly/blob/61dac80e3b9b83d81b3fb783e6fdc1a865a81d06/tests/mobly/controllers/android_device_lib/adb_test.py#L630)):*
<details><summary><i>Previously, mhaoli (Minghao Li) wrote…</i></summary><blockquote>

Thanks for updating the test case test_forward. 

How about using args that are similiar to real forward command and reverse command in the test cases?

e.g.
adb.AdbProxy().forward(['tcp:12345', 'tcp:23456'])
adb.AdbProxy().reverse(['tcp:12345', 'tcp:23456'])
</blockquote></details>

Done.
___
*[`tests/mobly/controllers/android_device_lib/adb_test.py` line 638 at r3](https://reviewable.io/reviews/google/mobly/901#-Nh_jfiE0p6V_oO03lBJ:-NhbO7sD-wncjmdde7Cf:b-896fix) ([raw file](https://github.com/google/mobly/blob/61dac80e3b9b83d81b3fb783e6fdc1a865a81d06/tests/mobly/controllers/android_device_lib/adb_test.py#L638)):*
<details><summary><i>Previously, mhaoli (Minghao Li) wrote…</i></summary><blockquote>

ditto here
</blockquote></details>

Done.

**Review by mhaoli — APPROVED** [review]

**Review by mhaoli — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: all files reviewed, 4 unresolved discussions (waiting on @xpconanfan)

**Review by mhaoli — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/901)* status: all files reviewed, 1 unresolved discussion (waiting on @xpconanfan)

**Comment by xpconanfan** [comment]
Please rebase and resolve the conflict so we can merge

**Review by ssachinsunder — COMMENTED** [review]
It seems like there was a switch in py formatting tool. How do I use the new tool (pyink) to ensure formatting is right?

> *[Reviewable](https://reviewable.io/reviews/google/mobly/901#-:-NhlW1zXBgecDubRzlLm:b-uc7ams)* status: 0 of 89 files reviewed, 1 unresolved discussion (waiting on @mhaoli and @xpconanfan)

**Comment by ssachinsunder** [comment]
Sorry closed this accidentally by discarding commits via GitHub UI (didn't expect that to happen). Created a new pull request - https://github.com/google/mobly/pull/903

**Pull request closed without merging**